airbyte-webapp/src/views/Connector/ConnectorCard/ConnectorCard.tsx

```diff
@@ -38,14 +41,16 @@ interface ConnectorCardBaseProps {
 
   // used in ConnectorCard and ConnectorForm
   formType: "source" | "destination";
+  /**
+   * id of the selected connector definition id - might be available even if the specification is not loaded yet
+   * */
+  selectedConnectorDefinitionId?: string | null;
```

Comment:
I tried to find a good cutoff point to stop with the cleanup. The null is coming from here: https://github.com/airbytehq/airbyte/blob/7d6f9770503d1d6f34697513529d0fa13e0040a3/airbyte-webapp/src/components/destination/DestinationForm/DestinationForm.tsx#L46

I thought it would be OK for now because the `jobId` is doing the same thing, but if you feel strongly about it I can  change it.